extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClientFallbackFactoryTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ *  Copyright (c) 2023 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Implement automatic Hashicorp Vault token renewal
+ *
+ */
+
+package org.eclipse.edc.vault.hashicorp;
+
+import okhttp3.Request;
+import org.eclipse.edc.spi.http.FallbackFactories;
+import org.junit.jupiter.api.Test;
+
+import static org.mockito.Mockito.mockStatic;
+
+class HashicorpVaultClientFallbackFactoryTest {
+
+    private static final int[] NON_RETRYABLE_STATUS_CODES = {200, 204, 400, 403, 404, 405};
+
+    @Test
+    void create_shouldInitializeWithCorrectStatusCodes() {
```

Comment:
I'm curious why you'd want to test the implementation of the httpClient.execute method inside the Hashicorp Vault Extension. As a caller, would you not expect it to work when you pass the Fallback Factory?

This ensures the correct set of http status codes. The usage of this fallback factory is verified inside the HashicorpVaultClient. I wouldn't go beyond that.

Reply:
This test is actually not verifying a behavior (that's already tested in the http client), but some magic values (the status code list).
So you won't be able to separate the values from the behavior (what's done in the httpclient test).
The alternative is to test this in an integration level, but it would be too effort for little value.

As I said I would delete this, but I could also get along, it's just philosophy of unit-testing :)